arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -49,37 +49,24 @@ class ContentTableViewController: UITableViewController {
         self.downloadProgressView = downloadProgressView
     }
 
-    // MARK: - Table view data source
-
-    override func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
-    }
+    // MARK: - UITableViewDataSource
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return displayedSamples.count
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let reuseIdentifier = "ContentTableCell"
-        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier) as! ContentTableCell
-
         let sample = displayedSamples[indexPath.row]
-        cell.titleLabel.text = sample.name
-        
-        if self.expandedRowIndex == indexPath.row {
-            cell.detailLabel.text = sample.description
-        } else {
-            cell.detailLabel.text = nil
-        }
-        
-        cell.infoButton.addTarget(self, action: #selector(ContentTableViewController.expandCell(_:)), for: .touchUpInside)
-        cell.infoButton.tag = indexPath.row
-
-        cell.backgroundColor = .clear
         
+        let cell = tableView.dequeueReusableCell(withIdentifier: "ContentTableCell") as! ContentTableCell
+        cell.titleLabel.text = sample.name
+        cell.detailLabel.text = sample.description
+        cell.isExpanded = expandedRowIndices.contains(indexPath.row)
```

Comment:
This is an interesting approach, but I think there might be a better one. One would be to use a subtitle cell and set the detail text label's text based one whether the cell should be expanded. That would be closer to what you have now, but without the need for a custom cell class. Another option would be to dequeue a basic cell or a subtitle cell based on whether the cell should be expanded. The advantage to that approach is that subviews are not being rearranged (potentially) each time a cell is loaded.

Reply:
In general I'd like that approach, but in this case I'd rather leave it as a custom subclass. I tried to use the detail cell in the prior redesign and received pushback for not having the control over vertical spacing that we wanted. Also, we'll need the subclass in order to achieve more advanced functionality in the future.